src/main/java/stef/projects/console/repository/UserRepositoryImpl.java

```diff
@@ -0,0 +1,199 @@
+        statement.setString(1, email);
+        ResultSet resultSet = statement.executeQuery();
+        while (resultSet.next()) {
+            showUser(resultSet);
+            quizAppUser = extractUser(resultSet.getLong("id"), resultSet.getString("first_name"), resultSet.getString("last_name"), resultSet.getString("username"), resultSet.getString("email"), resultSet.getString("password"), resultSet.getString("value"));
+        }
+        return quizAppUser;
+    }
+
+    @Override
+    public User findUsernameAndPassword(String username, String password) throws SQLException { //find or validate
+        System.out.println("SELECT_BY_USERNAME_AND_PASSWORD_QUERY");
+        statement = dbConnection(SELECT_BY_USERNAME_AND_PASSWORD_QUERY);
+        statement.setString(1, username);
+        statement.setString(2, password);
+        ResultSet resultSet = statement.executeQuery();
+        while (resultSet.next()) {
+            showUser(resultSet);
+            quizAppUser = extractUser(resultSet.getLong("id"), resultSet.getString("first_name"), resultSet.getString("last_name"), resultSet.getString("username"), resultSet.getString("email"), resultSet.getString("password"), resultSet.getString("value"));
+        }
+        return quizAppUser;
+    }
+
+    @Override
+    public List<User> findUsersByUserRole(UserRole userRole) throws SQLException {
+        System.out.println("SELECT_BY_USER_ROLE_QUERY");
+        statement = dbConnection(SELECT_BY_USER_ROLE_QUERY);
+        statement.setString(1, UserRole.getValue(userRole));
+        ResultSet resultSet = statement.executeQuery();
+        while (resultSet.next()) {
+            showUser(resultSet);
+            quizAppUser = extractUser(resultSet.getLong("id"), resultSet.getString("first_name"), resultSet.getString("last_name"), resultSet.getString("username"), resultSet.getString("email"), resultSet.getString("password"), resultSet.getString("value"));
+            quizAppUserList.add(quizAppUser);
+        }
+        return quizAppUserList;
+    }
+
+    @Override
+    public List<User> findUserByFirstNameOrLastName(String fistName, String lastName) throws SQLException {
+        System.out.println("SELECT_BY_FIRST_OR_LAST_NAME_QUERY");
+        statement = dbConnection(SELECT_BY_FIRST_OR_LAST_NAME_QUERY);
+        statement.setString(1, fistName);
+        statement.setString(2, lastName);
+        ResultSet resultSet = statement.executeQuery();
+        while (resultSet.next()) {
+            showUser(resultSet);
+            quizAppUser = extractUser(resultSet.getLong("id"), resultSet.getString("first_name"), resultSet.getString("last_name"), resultSet.getString("username"), resultSet.getString("email"), resultSet.getString("password"), resultSet.getString("value"));
+            quizAppUserList.add(quizAppUser);
+        }
+        return quizAppUserList;
+    }
+
+    private void extractStatement(User user, boolean update) throws SQLException {
```

Comment:
Extract is meant to return something so this method should be renamed to fillPreparedStatement
Also update should be renamed to isForUpdate or shouldUpdate
Finally in the arguments we should also pass the PreparedStatement reference